sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,968 @@
+        return Lists.transform(inputSplits,
+            new Function<SerializableSplit, BoundedSource<KV<K, V>>>() {
+              @Override
+              public BoundedSource<KV<K, V>> apply(SerializableSplit serializableInputSplit) {
+                HadoopInputFormatBoundedSource<K, V> hifBoundedSource =
+                    new HadoopInputFormatBoundedSource<K, V>(conf, keyCoder, valueCoder,
+                        keyTranslationFunction, valueTranslationFunction, inputFormatObj,
+                        serializableInputSplit);
+                return hifBoundedSource;
+              }
+            });
+      } else {
+        LOG.info("Not splitting source {} because source is already split.", this);
+        return ImmutableList.of((BoundedSource<KV<K, V>>) this);
+      }
+    }
+
+    @Override
+    public long getEstimatedSizeBytes(PipelineOptions po) throws Exception {
+      if (inputSplit == null) {
+        if (inputSplits == null) {
+          computeSplits();
+        }
+        return boundedSourceEstimatedSize;
+      }
+      return inputSplit.getSplit().getLength();
+    }
+
+
+    /**
+     * This is helper function to compute splits. This method will also calculate size of the data
+     * being read. Note : This method is called exactly once, the splits are retrieved and cached
+     * for further use by splitIntoBundles() and getEstimatesSizeBytes().
+     * @throws InterruptedException
+     */
+    @VisibleForTesting
+    void computeSplits() throws IOException, InterruptedException{
+      inputFormatObj = createInputFormat();
+      List<InputSplit> splits =
+          inputFormatObj.getSplits(Job.getInstance(conf.getHadoopConfiguration()));
+      if (splits == null) {
+        throw new IOException(HadoopInputFormatIOContants.COMPUTESPLITS_NULL_GETSPLITS_ERROR_MSG);
+      }
+      if (splits.isEmpty()) {
+        throw new IOException(HadoopInputFormatIOContants.COMPUTESPLITS_EMPTY_SPLITS_ERROR_MSG);
+      }
+      boundedSourceEstimatedSize = 0;
+      inputSplits = new ArrayList<SerializableSplit>();
+      for (InputSplit inputSplit : splits) {
+        if (inputSplit == null) {
+          throw new IOException(HadoopInputFormatIOContants.COMPUTESPLITS_NULL_SPLIT_ERROR_MSG);
+        }
+        boundedSourceEstimatedSize = boundedSourceEstimatedSize + inputSplit.getLength();
+        inputSplits.add(new SerializableSplit(inputSplit));
+      }
+      validateKeyValueClasses();
+    }
+
+    /**
+     * Returns instance of InputFormat set in the configuration.
+     */
+    private InputFormat<?, ?> createInputFormat() throws IOException, InterruptedException {
+      InputFormat<?, ?> inputFormatObj;
+      try {
+        inputFormatObj = (InputFormat<?, ?>) conf.getHadoopConfiguration()
+            .getClass(HadoopInputFormatIOContants.INPUTFORMAT_CLASSNAME, null).newInstance();
+        /**
+         * Sets configuration if provided InputFormat implements Configurable.
```

Comment:
Added more explanation here, the comment is important, as it is specially for the case when Input format explicitly implements Configurable interface . Added the following comment : "  /**
           * If InputFormat explicitly implements interface {@link Configurable}, then setConf()
           * method of {@link Configurable} needs to be explicitly called to set all the
           * configuration parameters. Example InputFormat classes which implement Configurable are
           * {@link org.apache.hadoop.mapreduce.lib.db.DBInputFormat DBInputFormat},
           * {@link org.apache.hadoop.hbase.mapreduce.TableInputFormat TableInputFormat}, etc.
           */ "